### PR TITLE
feat: Dns forwarder with load balancer and virtual machine scale set

### PR DESCRIPTION
**Pull request by umbcoppolabottazzi**
### List of changes

Added a new module for the DNS forwarder that includes:

- Virtual Machine Scale Set with a custom image
- Load Balancer with a static IP
- Key Vault for storing the password of the Virtual Machine Scale Set
- Network Security Group for the Virtual Machine Scale Set
- [Optional]: Subnet for the Virtual Machine Scale Set + Subnet for the Load Balancer

### Type of changes

- [x] Add new module
- [ ] Update existing module
- [ ] Remove existing module

### Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### Run checks

Useful commands to run checks on local machine

```sh
bash .utils/terraform_run_all.sh init local
pre-commit run -a
```

### CODE REVIEW

**Review by mamari90 — APPROVED** [review]

File: dns_forwarder_lb_vmss/main.tf [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  virtual_network_name = var.virtual_network_name
+  address_prefixes     = [var.address_prefixes_lb]
+}
+
+#
+# Network security group - virtual machine scale set
+#
+
+resource "azurerm_network_security_group" "vmss" {
+  count = var.subnet_vmss_id != null ? 0 : 1
```
Comment (diegolagospagopa) [inline dns_forwarder_lb_vmss/main.tf:47]: Is correct that is not created is the subnet is external?, May be i created the subnet but i don't know anything about the NSG

File: dns_forwarder_lb_vmss/main.tf [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    source_address_prefix      = "*"
+    destination_address_prefix = "*"
+  }
+}
+
+#
+# Load balancer
+#
+
+module "lb" {
```
Comment (diegolagospagopa) [inline dns_forwarder_lb_vmss/main.tf:95]: Please can you change the name in


```suggestion
module "load_balancer" {
```

File: dns_forwarder_lb_vmss/main.tf [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+resource "random_password" "psw" {
+  length           = 20
+  special          = true
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+
+#tfsec:ignore:azure-keyvault-ensure-secret-expiry
+resource "azurerm_key_vault_secret" "psw_vmss" {
+  name         = local.prefix
```
Comment (diegolagospagopa) [inline dns_forwarder_lb_vmss/main.tf:184]: Please give a more complex name to be more readable

File: dns_forwarder_lb_vmss/variables.tf [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+variable "object_id_group_ad" {
+  type        = string
+  description = "(Optional) The ID of the group object that can read/write access to the kv."
+  default     = null
+}
+
+variable "storage_sku" {
+  type        = string
+  description = "(Optional) The SKU of the storage account with which to persist VM. Use a singular sku that would be applied across all disks, or specify individual disks. Usage: [--storage-sku SKU | --storage-sku ID=SKU ID=SKU ID=SKU...], where each ID is os or a 0-indexed lun. Allowed values: Standard_LRS, Premium_LRS, StandardSSD_LRS, UltraSSD_LRS, Premium_ZRS, StandardSSD_ZRS."
+  default     = "StandardSSD_LRS"
```
Comment (diegolagospagopa) [inline dns_forwarder_lb_vmss/variables.tf:87]: Please change into


```suggestion
  default     = "StandardSSD_ZRS"
```

**Review by diegolagospagopa — CHANGES_REQUESTED** [review]
.

**Review by diegolagospagopa — APPROVED** [review]
.

**Pull request merged**